ps_facetedsearch.php

```diff
@@ -1595,4 +1651,67 @@ public function getFilterTemplate($idFilterTemplate)
             WHERE id_layered_filter = ' . (int) $idFilterTemplate
         );
     }
+
+    /**
+     * Returns array with all controllers supported by this module
+     */
+    public function getSupportedControllers()
+    {
+        return $this->supportedControllers;
+    }
+
+    public function setSupportedControllers($supportedControllers)
+    {
+        $this->supportedControllers = $supportedControllers;
+    }
+
+    /**
+     * Returns array with all controllers supported by this module
+     */
+    public function isControllerSupported($controller)
+    {
+        return isset($this->supportedControllers[$controller]);
+    }
+
+    /**
+     * Should this controller filter blocks be cached?
+     */
+    public function shouldCacheController($controller)
+    {
+        return $this->supportedControllers[$controller]['cacheable'];
+    }
+
+    public function initializeSupportedControllers()
```

Comment:
It probably doesn't need to be an initializer. Why don't just have `getSupportedControllers(): array`? Later on, in a different PR, we could think about hooks to make it flexible.